c2corg_ui/static/js/map/baselayerselector.js

```diff
@@ -0,0 +1,263 @@
+goog.provide('app.BaselayerSelectorController');
+goog.provide('app.baselayerSelectorDirective');
+
+goog.require('app');
+goog.require('ngeo.BackgroundLayerMgr');
+goog.require('ol.Attribution');
+goog.require('ol.format.WMTSCapabilities');
+goog.require('ol.layer.Tile');
+goog.require('ol.source.OSM');
+goog.require('ol.source.WMTS');
+goog.require('ol.tilegrid.WMTS');
+
+
+/**
+ * This directive is used to display a map baselayer selector.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.baselayerSelectorDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppBaselayerSelectorController',
+    controllerAs: 'bglayerCtrl',
+    bindToController: true,
+    scope: {
+      'map': '=appBaselayerSelectorMap'
+    },
+    templateUrl: '/static/partials/map/baselayerselector.html'
+  };
+};
+
+
+app.module.directive('appBaselayerSelector', app.baselayerSelectorDirective);
+
+
+/**
+ * @param {angular.$http} $http
+ * @param {ngeo.BackgroundLayerMgr} ngeoBackgroundLayerMgr Background layer
+ *     manager.
+ * @param {string} ignApiKey IGN API key.
+ * @param {app.Alerts} appAlerts
+ * @param {app.Authentication} appAuthentication
+ * @constructor
+ * @export
+ * @struct
+ * @ngInject
+ */
+app.BaselayerSelectorController = function($http, ngeoBackgroundLayerMgr,
+  ignApiKey, appAlerts, appAuthentication) {
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {ngeo.BackgroundLayerMgr}
+   * @private
+   */
+  this.backgroundLayerMgr_ = ngeoBackgroundLayerMgr;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.ignApiKey_ = ignApiKey;
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.alerts_ = appAlerts;
+
+  /**
+   * @type {ol.Map}
+   * @export
+   */
+  this.map;
+
+  /**
+   * @type {Object}
+   * @private
+   */
+  this.cachedLayers_ = {};
+
+  /**
+   * @type {ol.format.WMTSCapabilities}
+   * @private
+   */
+  this.wmtsParser_;
+
+  /**
+   * @type {Object}
+   * @export
+   */
+  this.currentBgLayer;
+
+  /**
+   * @type {Array.<Object>}
+   * @export
+   */
+  this.bgLayers = app.BaselayerSelectorController.BACKGROUND_LAYERS.filter(
+    function(layer) {
+      return !('auth' in layer) || !layer['auth'] ||
+        (layer['auth'] && appAuthentication.isAuthenticated());
```

Comment:
I don't think so because you could have:
* no 'auth' property set => your test would evaluate to false I guess whereas if there is no 'auth' it means the layer is public
* it could be set to 'false' (meaning it's public)